lc0/src/neural/network_cudnn.cu

```diff
@@ -0,0 +1,836 @@
+/*
+  This file is part of Leela Chess Zero.
+  Copyright (C) 2018 The LCZero Authors
+
+  Leela Chess is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  Leela Chess is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with Leela Chess.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "network_cudnn.h"
+#include "utils/bititer.h"
+#include <functional>
+#include <cassert>
+#include <mutex>
+
+#include <cublas_v2.h>
+#include <cudnn.h>
+
+
+void cudnnError(cudnnStatus_t status, const char* file, const int& line)
+{
+    if (status != CUDNN_STATUS_SUCCESS)
+    {
+        printf("CUDNN error: %s (%s:%d)\n", cudnnGetErrorString(status), file, line);
+        getchar();
+        exit(1);
+    }
+}
+
+#define reportCUDNNErrors(status)  cudnnError(status, __FILE__, __LINE__)
+
+
+// 256 MB fixed scratch memory size (hardcoded for now)
+#define CUDA_SCRATCH_SIZE (256*1024*1024)
+
+// hard-coded for now, no point in going above this anyway (can possibly save memory by reducing this)
+static constexpr int MAX_BATCH_SIZE = 1024;
```

Comment:
I used google style guide for other C++ files. https://google.github.io/styleguide/cppguide.html
Would it be possible to change this file to use it too, so that file format is consistent.

Particularly:
Names of global constants are in format kMaxBatchSize.

Multi-word variable name are in this format: multi_word_name
Class data members have trailing underscore:  in_tensor_desc_

Most of other requirements can be fixed automatically by clang-format (I can run it later on checked-in code)

Reply:
1024 seems very big for maximum batch size. I don't think low end cards even have enough memory to fit this many batches. This too should be probably allocated dynamically to avoid wasting huge amounts of GPU memory.

Reply:
updated most names, except for ones in small util classes (layer implementations).